test/source/browser/controllable.ts

```diff
@@ -54,41 +54,29 @@ abstract class ControllableBase {
     const selectors = this.selsAsProcessedArr(selector);
     this.log(`wait_all:1:${selectors.join(',')}`);
     for (const selector of selectors) {
+      // ignore visibility for at this stage as we don't care if this element is scrolled to
       this.log(`wait_all:2:${selector}`);
       if (this.isXpath(selector)) {
         this.log(`wait_all:3:${selector}`);
-        await this.target.waitForXPath(selector, { timeout: timeout * 1000, visible });
+        await this.target.waitForXPath(selector, { timeout: timeout * 1000 });
         this.log(`wait_all:4:${selector}`);
       } else {
         this.log(`wait_all:5:${selector}`);
-        await this.target.waitForSelector(selector, { timeout: timeout * 1000, visible });
+        await this.target.waitForSelector(selector, { timeout: timeout * 1000 });
```

Comment:
@tomholub What would you say about this fix?
Puppeteer had a breaking change and the element isn't considered `visible` now if it's below the scrolled area (as I understood):
https://github.com/puppeteer/puppeteer/blob/c21c3ba45aa9f7aba21eb33a9b3faac767f0e3e1/src/injected/util.ts#L72
In our tests we consider the element visible if it can be scrolled to.
Luckily, in our `waitAny` method we have a different mechanism of determining presence or visibility of the element, so the easiest was to combine the two.
Perhaps, I need to add more comments to the code regarding the `visible` parameter, as it is becoming our own custom concept.
`visible: true` -- element is present and visible (or will become visible when scrolled to)
`visible: false` -- element is present but deliberately hidden
`visible: undefined` -- element is present in whatever status.

Reply:
PS: Looks like this is the only place in tests where `visible: false` is used in tests in `flowcrypt-browser`
https://github.com/FlowCrypt/flowcrypt-browser/blob/ce108ca9b2021e7b4867c64910e4e6605985ba72/test/source/tests/gmail.ts#L397

And in `waitAny` calls `visible: false` is treated in the same way as `visible: undefined`.
So we need to update this method too somehow to unify the concept.
Maybe forbid `visible: false` as argument to `waitAny`, only allow `true | undefined`

Reply:
I'll let you use your judgement here, have no issue with it.